pkg/bootstrap/bootstrap.go

```diff
@@ -287,77 +283,94 @@ func setChartValues(manifestsDir string, nodeConfig *daemonconfig.Node, cfg cmds
 
 	var errs []error
 	for fileName, info := range files {
-		if err := rewriteChart(fileName, info, chartValues, serializer); err != nil {
+		if err := rewriteChart(fileName, info, chartValues); err != nil {
 			errs = append(errs, err)
 		}
 	}
 	return merr.NewErrors(errs...)
 }
 
 // rewriteChart applies cluster configuration values to the file at fileName with associated info.
-// If the file cannot be decoded as a HelmChart, it is silently skipped. Any other IO error is considered
-// a failure.
-func rewriteChart(fileName string, info os.FileInfo, chartValues map[string]string, serializer *json.Serializer) error {
-	b, err := ioutil.ReadFile(fileName)
+func rewriteChart(fileName string, info os.FileInfo, chartValues map[string]string) error {
+	fh, err := os.OpenFile(fileName, os.O_RDWR, info.Mode())
 	if err != nil {
-		return errors.Wrapf(err, "Failed to read manifest %s", fileName)
+		return errors.Wrapf(err, "failed to open manifest %s", fileName)
 	}
+	defer fh.Close()
 
 	// Ignore manifest if it cannot be decoded
-	obj, _, err := serializer.Decode(b, nil, nil)
+	objs, err := yaml.ToObjects(fh)
 	if err != nil {
-		logrus.Debugf("Failed to decode manifest %s: %s", fileName, err)
+		logrus.Warnf("Failed to decode manifest %s: %s", fileName, err)
 		return nil
 	}
 
-	// Ignore manifest if it is not a HelmChart
-	chart, ok := obj.(*helmv1.HelmChart)
-	if !ok {
-		logrus.Debugf("Manifest %s is %T, not HelmChart", fileName, obj)
-		return nil
-	}
+	var changed bool
 
-	// Generally we should avoid using Set on HelmCharts since it cannot be overridden by HelmChartConfig,
-	// but in this case we need to do it in order to avoid potentially mangling the ValuesContent field by
-	// blindly appending content to it in order to set values.
-	if chart.Spec.Set == nil {
-		chart.Spec.Set = map[string]intstr.IntOrString{}
-	}
+OBJECTS:
```

Comment:
What does this line do? Are we using a `goto` statement somewhere else to reference this?

Reply:
See a few lines below where if the `spec.set` field is improperly formatted, we break out of processing this object instead of failing over and over again for every key.